django/cantusdb_project/main_app/views/views.py

```diff
@@ -959,6 +966,36 @@ def redirect_documents(request) -> HttpResponse:
     return redirect(new_path)
 
 
+def redirect_chant_list(request) -> HttpResponse:
+    source_id: str = request.GET.get("source")
+    if source_id is None:
+        # source parameter must be provided
+        raise BadRequest("Source parameter must be provided")
+
+    base_url: str = reverse("chant-list", args=[source_id])
+
+    # optional search params
+    feast_id: str = request.GET.get("feast")
```

Comment:
These should all be `Optional[str]`

```suggestion
    feast_id: Optional[str] = request.GET.get("feast")
```

A good rule is that if you need to check if it's `None` later, you must have marked it as Optional.

(Are you using any sort of type-checker, like mypy, when you develop? It helps catch these things. Also a good IDE will flag any unused imports, etc.)